File: 1-two-sum/two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    const numMap = new Map();

    for(let i = 0; i < nums.length; i++) {
        let compliment = target - nums[i];

        if(numMap.has(compliment)){
            return [numMap.get(compliment), i];
        }

        numMap.set(nums[i], i);
    }
};

File: 1-two-sum/two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        nmap = {}

        for i in range(len(nums)):
            compliment = target - nums[i]
            if compliment in nmap:
                return [nmap[compliment], i]
            nmap[nums[i]] = i